Added comments, cleaned up the code a bit and updated README

paulpela opened the pull request.

marekpridal commented:
Thanks a lot for contribution! I'll take a look as soon as possible 🙂

marekpridal commented on Sources/BarChartKit/SwiftUI/BarChart.swift:
```diff
@@ -1,40 +1,38 @@
+    /// Provides data for the BarChart
     public struct DataSet {
+        /// Single data element for BarChart. Can consist of multiple bars
         public struct DataElement: Equatable, Identifiable {
```
Move please conformance of protocols to extensions where implementation is.

marekpridal commented on Sources/BarChartKit/SwiftUI/BarChart.swift:
```diff
@@ -104,6 +110,30 @@ public struct BarChart: View {
     }
 }
 
+extension BarChart.DataSet.DataElement {
```
Make extensions public, too.

marekpridal reviewed (changes requested):
Thanks a lot for improving documentation and code cleanest! I added few comments. Also I noticed GitHub Actions seems to be disabled in you profile and because of that the CI check cannot pass. Can you please enable it to be sure, everything works as expected?

paulpela commented:
Hi, I fixed conformance locations and added public wherever needed, however I can't fix the checks issue. Actions are definitely enabled on my repository. Any ideas?

paulpela commented:
Looks like it worked now! Sorry for the mess! I learned a lot here :)

marekpridal reviewed (approved):
Thanks a lot for update!
